pulp_ansible/app/serializers.py

```diff
@@ -764,6 +967,57 @@ class Meta:
         )
 
 
+class CollectionVersionSigstoreSignatureSerializer(NoArtifactContentUploadSerializer):
+    """
+    A serializer for Sigstore signature models.
+    """
+    sigstore_bundle = serializers.JSONField(
+        help_text=_("A Sigstore bundle used for offline verification."),
+        allow_null=True,
+    )
+    signed_collection = DetailRelatedField(
+        help_text=_("The content this signature is pointing to."),
+        view_name_pattern=r"content(-.*/.*)-detail",
+        queryset=CollectionVersion.objects.all(),
+    )
+    sigstore_signing_service = RelatedField(
+        help_text=_("The signing service used to sign the collections"),
+        view_name="sigstore-signing-services-detail",
+        read_only=True,
+        allow_null=True,
+    )
+
+    def __init__(self, *args, **kwargs):
+        """Ensure `file` field is required."""
+        super().__init__(*args, **kwargs)
+        self.fields["file"].required = True
+
+    def validate(self, data):
+        """
+        Verify the signature is valid before creating it.
+        """
+        if not "repository" in data:
+            raise serializers.ValidationError(
+                "'repository' field not set"
+            )
+        data = super().validate(data)
+
+        if "request" not in self.context:
+            # Validate is called twice, first on the viewset, and second on the create task
+            # data should be set up properly on the second time, when request isn't in context
+            data = verify_sigstore_signature_upload(data)
+
+        return data
```

Comment:
This way of performing secondary validation is outdated now. 
```suggestion
    def deferred_validate(self, data):
        data = super().deferred_validate(data)
        data = verify_sigstore_signature_upload(data)
        return data
```